code-of-conduct.md

```diff
@@ -0,0 +1,80 @@
+# Contributor Covenant Code of Conduct
+
+## Our Pledge
+
+In the interest of fostering an open and welcoming environment, we as
+contributors and maintainers pledge to making participation in our project and
+our community a harassment-free experience for everyone, regardless of age, body
+size, disability, ethnicity, sex characteristics, gender identity and expression,
+level of experience, education, socio-economic status, nationality, personal
+appearance, race, religion, or sexual identity and orientation.
+
+## Our Standards
+
+Examples of behavior that contributes to creating a positive environment
+include:
+
+* Using welcoming and inclusive language
+* Being respectful of differing viewpoints and experiences
+* Gracefully accepting constructive criticism
+* Focusing on what is best for the community
+* Showing empathy towards other community members
+
+Examples of unacceptable behavior by participants include:
+
+* The use of sexualized language or imagery and unwelcome sexual attention or
+  advances
+* Trolling, insulting/derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or electronic
+  address, without explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+This list is not meant to be exhaustive, but merely to enumerate some common
+kinds of behaviors.
+
+## Our Responsibilities
+
+Project maintainers are responsible for clarifying the standards of acceptable
+behavior and are expected to take appropriate and fair corrective action in
+response to any instances of unacceptable behavior.
+
+Project maintainers have the right and responsibility to remove, edit, or
+reject comments, commits, code, wiki edits, issues, and other contributions
+that are not aligned to this Code of Conduct, or to ban temporarily or
+permanently any contributor for other behaviors that they deem inappropriate,
+threatening, offensive, or harmful.
+
+## Scope
+
+This Code of Conduct applies both within project spaces and in public spaces
+when an individual is representing the project or its community. Examples of
+representing a project or community include using an official project e-mail
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at [INSERT EMAIL ADDRESS]. All
```

Comment:
We need to specify a way to contact us for reporting unacceptable behavior. I know the docs are hosted on `*.dask.org`, is it possible to create an e-mail address like `conduct@dask.org`?

Reply:
We have `daskdevelopers@gmail.com`, which is used to manage a few shared services. Myself, Matt, and a few others currently have access. Perhaps `daskdevelopers+conduct@gmail.com` (gmail ignores `+...` in email addresses, but still allows filtering on it), which would allow filtering conduct emails into a folder, as well as auto-ccing the personal emails of those who should be notified (so people don't need to directly check this email address).

Reply:
Using `daskdevelopers+conduct@gmail.com` seems reasonable to me.

Also, good to learn that about gmail! I had no idea about that feature

Reply:
I think that we should make it clear which individuals will and will not receive an e-mail.  There should probably also be a way to submit a concern anonymously.  

This might make people feel more comfortable submitting something, particularly if it is about someone that they expect to be on the receiving end of that e-mail.

Reply:
That seems like a good idea. SciPy does this as well.

ref: https://docs.scipy.org/doc/scipy/reference/dev/conduct/code_of_conduct.html#reporting-guidelines

Reply:
> There should probably also be a way to submit a concern anonymously

Agreed. I think an online form (e.g. using Google forms) would work well for this. Jupyter uses [this form](https://docs.google.com/forms/d/e/1FAIpQLSfUuaGbXrH0anroNIkbc-dBrn-ISsC_BA57IRt57aQW85nK6w/viewform?c=0&w=1) for incident reporting, where things like name and e-mail are optional to allow for anonymous reporting. I'm in favor of adopting something similar to that. 

> we should make it clear which individuals will and will not receive an e-mail

I also agree with this point. This brings us to the question: who should receive these e-mails? Presumably this will be the group responsible for handling reported COC concerns. 

On a related note, it was proposed on a previous monthly developer call to use a separate e-mail account for conduct-related matters (e.g. something like `dask-conduct@googlegroups.com`). This will help us avoid any unnecessary overlap between those who receive `daskdevelopers@gmail.com` e-mails and those who will have access to the COC reporting account.